### ISSUE TITLE
Bug in Intro to TensorFlow lab, accuracy=0.1 and no Loss

### ISSUE THREAD

**Issue by charlie-wartnaby** [udacity/sdc-issue-reports#894]
A few people have encountered getting an empty graph frame for the Loss plot, and accuracy sticking at 0.1, so the lab didn't work. After some investigation, this turned out to be because of a maths exception (taking the log of zero or -ve) when computing the cross-entropy. Saturating the input value to be finite worked, but the moderator found a better-looking function:  tf.nn.softmax_cross_entropy_with_logits . See detail on the forum here: https://discussions.udacity.com/t/accuracy-0-10-in-the-intro-to-tensorflow-lab/272469

**Comment by subodh-malgonde** [comment]
Just to add to what Charlie has said, the lab notebook uses `-tf.reduce_sum(labels * tf.log(prediction), axis=1)`, which does not handle numerically unstable corner cases. This might sometimes lead to empty loss plots.

**Comment by baumanab** [comment]
Thanks for reporting this.  I'm a bit rusty on Tensorflow tt sounds like this ```tf.reduce_sum(labels * tf.log(prediction), axis=1)``` should be replaced using this function ```tf.nn.softmax_cross_entropy_with_logits()```.  What is the proposed final loss code?

**Comment by charlie-wartnaby** [comment]
Yes you're right, not as rusty as you think perhaps! I noted that a moderator pointed me to  tf.nn.softmax_cross_entropy_with_logits(), though I had already put in my own workaround to replace any zeroes with a finite epsilon value, I think.

**Comment by baumanab** [comment]
Thanks again for helping with this correction.  A note has been added to the lab code to direct students to the alternative cross entropy function.
